src/MongoDB.Driver.Core/Core/Compression/SnappyCompressor.cs

```diff
@@ -29,14 +30,22 @@ internal class SnappyCompressor : ICompressor
 
         public void Compress(Stream input, Stream output)
         {
-#if NET452 || NETSTANDARD2_0
-            using (var snappyStream = new SnappyStream(output, CompressionMode.Compress, true))
+            using (var temp = new MemoryStream())
             {
-                input.EfficientCopyTo(snappyStream);
-            }
+#if NET452 || NETSTANDARD2_0
+                using (var snappyStream = new SnappyStream(temp, CompressionMode.Compress, true))
+                {
+                    input.EfficientCopyTo(snappyStream);
+                }
+                var newBytes = temp.ToArray().Skip(18).ToArray();
```

Comment:
The only new logic here is to remove first 18 bytes from a request to the server. These 18 bytes contain unexpected headers. See the test below.

Reply:
**NOTE**: This code doesn't pretend to be well-coded or has a good performance.
**NOTE2**: This code is processed by a server. And for this case, a server returns a response for decompression